rpcs3/Emu/rpcn_config.cpp

```diff
@@ -0,0 +1,75 @@
+#include "stdafx.h"
+#include "rpcn_config.h"
+#include "Emu/System.h"
+
+cfg_rpcn g_cfg_rpcn;
+
+void cfg_rpcn::load()
+{
+	if (fs::file cfg_file{cfg_rpcn::get_path(), fs::read})
+	{
+		from_string(cfg_file.to_string());
+	}
+}
+
+void cfg_rpcn::save() const
+{
+	fs::file(cfg_rpcn::get_path(), fs::rewrite).write(to_string());
+}
+
+std::string cfg_rpcn::get_path()
+{
+	return fs::get_config_dir() + "rpcn.yml";
+}
+
+std::string cfg_rpcn::generate_npid()
+{
+	std::string gen_npid = "RPCS3_";
+
+	const char list_chars[] = {'0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X',
+	    'Y', 'Z', 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z'};
+
+	std::srand(time(0));
+
+	for (int i = 0; i < 10; i++)
+	{
+		gen_npid += list_chars[std::rand() % (sizeof(list_chars) - 1)];
```

Comment:
I am aware that subtracting 1 actually truncates very last character 'z'. Behavior of % operator does the following for example _[I suppose size of list_char is 64 (27+27+10)]_: 
```
rand() gives a number [0..32768]
% (64 - 1) for rand() = 64 gives 1 which is incorrect
% 64 for rand() = 64 gives 0, which is correct. 64th element from this circular array is the very first.
```
```suggestion
		gen_npid += list_chars[std::rand() % (sizeof(list_chars))];
```